rfcs/065-library-data-link-explorer/README.md

```diff
@@ -0,0 +1,128 @@
+3. **Frontend displays the graph**: The frontend renders a visual graph of nodes (works) and edges (relationships), allowing users to explore the connections using a graph library.
+4. **Enhanced Collections Information workflow**: This allows Collections Information I colleagues to independently view and explore relationships between records without requiring software engineer assistance, boosting productivity and collaboration.
+
+---
+
+## Backend
+
+- An API endpoint, powered by AWS Lambda to:
+    - query Elasticsearch for works, returning matched works of the given work id
+    - generate a DOT notation graph for this given work id using matcher logic within the endpoint
+    - transform this DOT notation into JSON ready for front end consumption
+- Accessed securely using AWS Secrets Manager for credentials for Elasticsearch
+
+### Helpers
+
+- Use the build of the content-api as an example API endpoint
+- JSON conversion is not strictly necessary, but useful for better graph libraries
+- [existing matcher logic](https://github.com/wellcomecollection/catalogue-pipeline/blob/main/pipeline/matcher_merger/matcher/scripts/getMatcherGraph.ts)
+
+### Technical Considerations
+
+- Frameworks and Dependencies
+    - TypeScript
+    - aws-lambda
+    - ts graphviz for generating dotnotation ready for conversion?
+    - AWS SDK for secrets - or is this done in Elasticsearch?
+    
+
+---
+
+## Frontend
+
+- A Next.js application that:
+    - provides a user interface for CI staff to enter a work id
+    - renders the matcher graph dynamically using a graph library (see candidates below)
+        - uses nodes and edges to show relationships
+    - handle error display when no relationships are present?
+    - potentially offer more advanced features than is currently possible (i.e, zoom, node highlighting
+- Accessed securely - consider later down the line
+
+---
+
+## Infrastructure and Deployment
+
+- Use GitHub Actions for CI/CD to deploy the backend and frontend
+- keeping repositories separate for clarity and modularity between the API/backend logic and frontend application and deployment
+
+---
+
+## Graph Library Candidates
+
+- https://js.cytoscape.org/
+- https://www.sigmajs.org/ + [ReactSigma](https://sim51.github.io/react-sigma/) for prebuilt components
+- https://github.com/visjs/vis-network
+- Graphviz wrappers - https://github.com/magjac/d3-graphviz, https://github.com/mdaines/viz-js
+
+---
+
+## Testing
+
+- Jest for backend logic
+- Playwright for API and frontend e2es
+
+---
+
+## Workflow
+
+1. API Development
+    1. using the content-api as a guide for structuring the api endpoint
+    2. implement the matcher logic within the endpoint
+        1. requires cleaning and configuring to both (work) and work with this setup
+    3. add transformation logic to convert dot to JSON
+    4. secure elasticsearch queries using Secrets Manager
+2. Frontend Application build
+    1. design base wireframe for working against
```

Comment:
Are you meant to get a hand with this? Could be super simple (one input field and a submit button, somewhere to render the graph or the error message) and then designed a bit more should it be something we want to develop further.

Reply:
This would be very useful - a lot of the graph libraries have some CSS *quirks* - requiring specific heights to be set in order to render the graph at all, for example. 

I've made a few "example" apps utilising an input and graph area.

In Sigma-React
```css
.SigmaContainer {
  width: 100%
  height: 900px; /* Graph won't render without this value */
};
```

I'm hoping even the most basic wireframe is there just to have something to work against, and make it consistent.